### PR TITLE
Feature/update delivery time cli 1119960948554225 joan

### PR DESCRIPTION
**Pull request by juanlopez4691**
### Ticket
- [Import Lieferzeiten / import delivery times](https://app.asana.com/0/inbox/383242129844222/1115912588841365/1119960948554225)

### CODE REVIEW

File: src/CliCommand.php [hunk 1]
```diff
@@ -17,39 +17,50 @@ class CliCommand extends \WP_CLI_Command {
...
-  public function refreshSalePercentage(array $args, array $options) {
-    $options = wp_parse_args($options, ['all' => '']);
-    if ($options['all']) {
-      $product_ids = wc_get_products([
-        'return' => 'ids',
-        'limit' => -1,
-      ]);
-      static::updateSalePercentage($product_ids);
-      WP_CLI::success('Sale percentage have been sucessfully updated.');
+  public function refreshSalePercentage(array $args, array $assoc_args) {
```
Comment (sun) [inline src/CliCommand.php]: The original signature `array $args, array $options` was correct.  A shell command consists of arguments and options.

File: src/CliCommand.php [hunk 2]
```diff
@@ -17,39 +17,50 @@ class CliCommand extends \WP_CLI_Command {
...
    * @param array $product_ids
-   *   List of product unique identifiers.
+   *   List of IDs of products to update.
+   * @param string $meta
+   *   Meta field to update.
    */
-  public static function updateSalePercentage($product_ids) {
+  public static function udpateProductsMeta($product_ids, $meta) {
     foreach ($product_ids as $product_id) {
       if (!$product_id = is_numeric($product_id) && (int) $product_id == $product_id ? (int) $product_id : 0) {
```
Comment (sun) [inline src/CliCommand.php]:
```suggestion
       if (!$product_id = is_numeric($product_id) && (int) $product_id === $product_id ? (int) $product_id : 0) {
```
I don't really understand what this condition is supposed to test though:
```
$ php -r 'var_dump($a = "123", $a == $a, (int) $a == $a, $a == (int) $a, (int) $a === $a, $a === (int) $a);'
string(3) "123"
bool(true)
bool(true)
bool(true)
bool(false)
bool(false)

$ php -r 'var_dump($a = "ab", $a == $a, (int) $a == $a, $a == (int) $a, (int) $a === $a, $a === (int) $a);'
string(2) "ab"
bool(true)
bool(true)
bool(true)
bool(false)
bool(false)
```

File: src/CliCommand.php (hunk 1 above)
Comment (LucaPipolo) [reply src/CliCommand.php]: On https://make.wordpress.org/cli/handbook/commands-cookbook/#annotating-with-phpdoc is  ` function hello( $args, $assoc_args ) {`.

I'd stick with what the offical documentation says.

Comment (sun) [reply src/CliCommand.php]: The documentation is wrong in this case. In addition, the original code used `$options` already.

**Comment by juanlopez4691** [comment]
@sun or @LucaPipolo Can you please review last fixes pushed?

File: src/CliCommand.php [hunk 3]
```diff
@@ -18,38 +18,45 @@ class CliCommand extends \WP_CLI_Command {
...
+   * Updates a given meta field for a list of products.
    *
    * @param array $product_ids
-   *   List of product unique identifiers.
+   *   List of IDs of products to update.
+   * @param string $meta
+   *   Meta field to update.
    */
-  public static function updateSalePercentage($product_ids) {
+  public static function udpateProductsMeta($product_ids, $meta) {
```
Comment (sun) [inline src/CliCommand.php]: Typo
```suggestion
  public static function updateProductsMeta($product_ids, $meta) {
```

File: src/CliCommand.php [hunk 4]
```diff
@@ -18,38 +18,45 @@ class CliCommand extends \WP_CLI_Command {
...
+      WP_CLI::error('A comma separated list of products IDs is needed. Use option --all to update all products.');
+    }
+  }
+
+  /**
+   * Updates products delivery time.
+   */
+  public function refreshDeliveryTime(array $args, array $options) {
+    if ($product_ids = getProductsToUpdate($args, $options)) {
+      static::udpateProductsMeta($product_ids, 'delivery_time');
```
Comment (sun) [inline src/CliCommand.php]: Typo
```suggestion
      static::updateProductsMeta($product_ids, 'delivery_time');
```

File: src/CliCommand.php [hunk 5]
```diff
@@ -18,38 +18,45 @@ class CliCommand extends \WP_CLI_Command {
...
-    $options = wp_parse_args($options, ['all' => '']);
-    if ($options['all']) {
-      $product_ids = wc_get_products([
-        'return' => 'ids',
-        'limit' => -1,
-      ]);
-      static::updateSalePercentage($product_ids);
-      WP_CLI::success('Sale percentage have been sucessfully updated.');
+    if ($product_ids = getProductsToUpdate($args, $options)) {
+      static::udpateProductsMeta($product_ids, 'sale_percentage');
```
Comment (sun) [inline src/CliCommand.php]: Typo
```suggestion
      static::updateProductsMeta($product_ids, 'sale_percentage');
```

File: src/CliCommand.php (hunk 3 above)
Comment (sun) [reply src/CliCommand.php]: The argument name `$meta` should be `$meta_key` for self-descriptive clarity.

**Pull request merged**